src/weapon/DlcWeaponData.ts

```diff
@@ -0,0 +1,93 @@
+import { WeaponHash } from '../hashes';
+import { getUInt32FromUint8Array } from '../utils/GetUInt32FromUInt8Array';
+import { getStringFromUInt8Array } from '../utils/GetStringFromBuffer';
+
+/**
+ * DlcWeaponData
+ * refer1: https://github.com/citizenfx/fivem/blob/master/code/client/clrcore/External/DlcWeaponStructs.cs#L10
+ * refer2: https://docs.fivem.net/natives/?_0xBF0FD6E56C964FCB
+ *
+ */
+export interface DlcWeaponData {
+  // int emptyCheck; //use DLC1::_IS_DLC_DATA_EMPTY on this
+  // int padding1;
+  // int weaponHash;
+  // int padding2;
+  // int unk;
+  // int padding3;
+  // int weaponCost;
+  // int padding4;
+  // int ammoCost;
+  // int padding5;
+  // int ammoType;
+  // int padding6;
+  // int defaultClipSize;
+  // int padding7;
+  // char nameLabel[64];
+  // char descLabel[64];
+  // char desc2Label[64]; // usually "the" + name
+  // char upperCaseNameLabel[64];
+  validCheck: number;
+  weaponHash: number;
+  unk: number;
+  weaponCost: number;
+  ammoCost: number;
+  ammoType: number;
+  defaultClipSize: number;
+  name: string;
+  desc: string;
+  simpleDesc: string;
+  upperCaseName: string;
+}
+
+/**
+ * DlcWeaponData
+ */
+export const DlcWeaponData = new Map<WeaponHash, DlcWeaponData>();
+
+/**
+ * Initialize DlcWeaponData, avoid calling expansive native repeatedly
+ *
+ */
+function initializeOnce() {
+  let isInitialized = false;
+
+  return function() {
+    if (isInitialized) {
+      return;
+    }
+
+    // magic number based on struct DlcWeaponData
+    const intLength = 4;
+    const strLength = 64;
+
+    const weaponCount = GetNumDlcWeapons();
+    for (let i = 0; i < weaponCount; i++) {
+      const buffer = new Uint8Array(14 * intLength + 4 * strLength);
+
+      // https://docs.fivem.net/natives/?_0x79923CD21BECE14E
+      Citizen.invokeNative('0x79923CD21BECE14E', i, buffer, Citizen.returnResultAnyway());
```

Comment:
Calling `GET_DLC_WEAPON_DATA` directly is faster than invoking the native. Please use `GetDlcWeaponData`

Reply:
I've tested many times and confirmed that calling `GET_DLC_WEAPON_DATA` directly would just crash the game :(. It seems the complexity of struct make an impact on that.